licenses/license.go

```diff
@@ -0,0 +1,267 @@
+		return fail(errors.New("license: not issued by " + licenseIssuer))
+	}
+
+	accountID, ok := claims[jwtAccountID].(float64)
+	if !ok || accountID < 0 {
+		return fail(errors.New("license: token contains no or invalid account ID"))
+	}
+	org, ok := claims[jwtOrganization].(string)
+	if !ok {
+		return fail(errors.New("license: token contains no organization"))
+	}
+	storageCap, ok := claims[jwtStorageCapacity].(float64)
+	if !ok {
+		return fail(errors.New("license: token contains no storage capacity"))
+	}
+	plan, ok := claims[jwtPlan].(string)
+	if !ok {
+		return fail(errors.New("license: token contains no plan"))
+	}
+
+	// The following fields are not present in older licenses.
+	// Hence, we cannot require them
+	licenseID, _ := claims[jwtLicenseID].(string)
+	isTrial, _ := claims[jwtTrial].(bool)
+
+	return License{
+		ID:           licenseID,
+		Organization: org,
+		AccountID:    uint64(accountID),
+		Plan:         plan,
+		StorageCap:   uint64(storageCap),
+		IssuedAt:     token.IssuedAt(),
+		ExpiresAt:    token.Expiration(),
+		NotAfter:     token.Expiration(),
+		Trial:        isTrial,
+	}, nil
+}
+
+// Verify checks whether a valid license is provided.
+//
+// Therfore, it first searches for a license in the following order:
+//  1. File referenced by MINIO_LICENSE env var, if any.
+//  2. A "./minio.license" file in the current working directory.
+//  3. The $HOME/"." + os.Args[0]/minio.license file if there are os.Args.
+//
+// If no license is present, Verify returns an error.
+// The license must also be issued after time.Now and,
+// in case of a trial license, must not be expired.
+// For non-trial licenses, a 30 day grace period is granted.
+func Verify() (License, error) {
+	fail := func(err error) (License, error) { return License{}, err }
+
+	var (
+		license []byte
+		err     error
+	)
+	if filename, ok := os.LookupEnv(licenseVar); ok && filename != "" {
+		license, err = os.ReadFile(filename)
+		if err != nil {
+			return fail(fmt.Errorf("license: %v", err))
+		}
+	}
+	if license == nil {
+		license, _ = os.ReadFile(licenseFile)
+	}
+	if license == nil && len(os.Args) > 0 {
+		home, err := os.UserHomeDir()
+		if err != nil {
+			return fail(fmt.Errorf("license: %v", err))
+		}
+
+		if license, err = os.ReadFile(filepath.Join(home, "."+os.Args[0], licenseFile)); err != nil {
+			return fail(fmt.Errorf("license: %v", err))
+		}
+	}
+	if license == nil {
+		return fail(errors.New("license: no license provided"))
+	}
+
+	l, err := Parse(string(license), jwt.WithKeySet(PublicKey()), jwt.UseDefaultKey(true), jwt.WithValidate(true), jwt.WithAcceptableSkew(licenseSkew))
+	if err != nil {
+		return fail(err)
+	}
+	if l.IssuedAt.After(time.Now()) {
+		return fail(errors.New("license: license is not yet valid"))
+	}
+	if l.Plan != planEnterpriseLite && l.Plan != planEnterprisePlus {
+		return fail(fmt.Errorf("license: either %s or %s plan required", planEnterpriseLite, planEnterprisePlus))
+	}
```

Comment:
In case of mineos , the legacy plans also need to be supported (`"TRIAL", "STANDARD", "ENTERPRISE"`)

Reply:
Really? So the behavior for a e.g. a "STANDARD" plan user would be that he can run mineos but not e.g. minwall. Are there users doing this @harshavardhana ?

If so we can change it to `license.Verify(plans ...string)`, I guess.